.travis.yml

```diff
@@ -44,24 +52,26 @@ install:
   - popd 
 
 before_script:
-  - sudo minikube start --vm-driver=none --cpus=1 --memory=1024
-  - sudo minikube update-context
-  # By default root is the owner of the .minikube directory. This is changed to $USER so that
-  # from the integration tests which run as a non-root user can access the files and connect
-  # to the minikube cluster
-  - sudo chown -R $USER $HOME/.minikube
-  - sudo chown -R $USER $HOME/.kube
+  - if [ "$XC_ARCH" == "amd64" ]; then
+      sudo minikube start --vm-driver=none --cpus=1 --memory=1024;
+      sudo minikube update-context;
+      sudo chown -R $USER $HOME/.minikube;
+      sudo chown -R $USER $HOME/.kube;
+      make shellcheck;
+    fi
 
 script:
   - make
   # Here sudo -E env "PATH=$PATH" make test is required for running tests with
   # sudo permissions since we are making use of scsi device mockdata in smartprobe_test
   # which could be fetched for a SCSI device with sudo permissions only since to access
   # a particular scsi device, sudo or root permissions are required.
-  - sudo -E env "PATH=$PATH" make test
-  - make integration-test
+  - if [ "$XC_ARCH" == "amd64" ]; then
+      sudo -E env "PATH=$PATH" make test;
+      make integration-test;
+      bash <(curl -s https://codecov.io/bash);
+    fi
 
 after_success:
```

Comment:
`after_success` should be called only after `script` in all jobs passes right? Otherwise if the testing fails in `amd64`, still image can be pushed for `arm64`

Reply:
above was mainly about test coverage data being pushed. i reverted this change as well.